### PR TITLE
Update Menu Button logic to account for default button

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
This PR updates the internal logic of Menu so that if a theme defines `theme.button.default`, then the Menu button will use the default button styling for its control as opposed to a `plain` button. The existing issue was that if Button has children, it acts as plain, so a Menu button was built by creating a Box (with direction row) and adding in the label/icon to appear like a Button. However, any button hover styles etc. no longer apply.

It makes more sense that the button control would not rely on children/plain (as this is a practice we want to avoid when possible) and instead relied on the `label` and `icon` props. This approach enhances the logic so that if a theme has `theme.button.default`, the menu control will inherit all the stylings of default button. This seemed like a more swift solution than try to extend all of the styling options into menu's theme object, particularly because Menu button was currently being implemented as a Box.

This new behavior applies only when `theme.button.default` is defined in the theme, but for backwards compatibility all of the previous logic remains as is.

#### Where should the reviewer start?
src/js/components/Menu/Menu.js

#### What testing has been done on this PR?
Tested `grommet-theme-hpe` locally in Menu stories. Added a jest test.

#### How should this be manually tested?
Apply a customTheme with `theme.button.default` styles defined (or use grommet-theme-hpe). Then, create a menu.

#### What are the relevant issues?
Related to https://github.com/grommet/hpe-design-system/issues/1199

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes, backwards compatible.

### CODE REVIEW

File: src/js/components/Menu/__tests__/Menu-test.js [hunk 1]
```diff
@@ -522,4 +535,16 @@ describe('Menu', () => {
     );
     expect(component.toJSON()).toMatchSnapshot();
   });
+
+  test('custom theme with default button', () => {
```
Comment (halocline) [inline src/js/components/Menu/__tests__/Menu-test.js:539]: Would it be helpful to build a snapshot test to cover the scenario where `button.default` is defined and the Menu has children? This would help ensure that the logic of rendering children if present continues to be applied.

File: src/js/components/Menu/Menu.js [hunk 2]
```diff
@@ -184,20 +184,30 @@ const Menu = forwardRef((props, ref) => {
...
+      ? (icon !== true && icon) || <MenuIcon color={iconColor} size={size} />
+      : null;
+
+  let content;
+  if (children) content = children;
+  else if (!theme.button.default) {
+    content = (
+      <Box
+        direction="row"
+        justify={justifyContent}
```
Comment (halocline) [inline src/js/components/Menu/Menu.js:199]: Per our convo, investigating the `justifyContent` + default button scenarion.

**Review by halocline — COMMENTED** [review]
Looking good. Two comments related to our discussion.

File: src/js/components/Menu/__tests__/Menu-test.js (hunk 1 above)
Comment (taysea) [reply src/js/components/Menu/__tests__/Menu-test.js:539]: Yes, I think you're right. We should have that as a baseline.

Comment (taysea) [reply src/js/components/Menu/__tests__/Menu-test.js:539]: Added this.

File: src/js/components/Menu/Menu.js (hunk 2 above)
Comment (taysea) [reply src/js/components/Menu/Menu.js:199]: An update here, currently button does not support ability to control justification of content within the width of the button. However, this could be considered a future enhancement. That being said, the benefits of `justifyContent` on Menu could still be achieved by providing Menu with children, such as:
```
<Menu>
        {() => (
          <Box align="center" gap="small" direction="row">
            <Text>Menu</Text>
            <FormDown />
          </Box>
        )}
</Menu>
```
This is not as seamless as a prop name, but the functionality is still possible. I feel like this is sufficient for now and a discussion of a button enhancement could be a future discussion. What do you think? @halocline

File: src/js/components/Menu/__tests__/Menu-test.js [hunk 3]
```diff
@@ -522,4 +535,27 @@ describe('Menu', () => {
...
```
Comment (ShimiSun) [inline src/js/components/Menu/__tests__/Menu-test.js:539]: Thanks for enhancing the testing! Can you try to separate the testing into a separate PR? After the testing PR will be merged it will be easier to understand the impact of changes this PR is suggesting.

Comment (taysea) [reply src/js/components/Menu/__tests__/Menu-test.js:539]: Yes, will do now.

File: src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap [hunk 4]
```diff
@@ -1401,11 +1401,20 @@ exports[`Menu custom theme with default button 1`] = `
   background: transparent;
   overflow: visible;
   text-transform: none;
-  outline: none;
   border: none;
-  padding: 0;
-  text-align: inherit;
-  color: inherit;
+  border-radius: 18px;
```
Comment (taysea) [inline src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap:1405]: These changes are expected because `default button` is now being used instead of `plain`

File: src/js/components/Menu/Menu.js [hunk 5]
```diff
@@ -184,20 +184,30 @@ const Menu = forwardRef((props, ref) => {
...
+        align="center"
+        pad="small"
+        gap={label && icon !== false ? 'small' : undefined}
+      >
+        {menuLabel}
+        {menuIcon}
+      </Box>
+    );
+  } else {
+    content = undefined;
```
Comment (taysea) [inline src/js/components/Menu/Menu.js:209]: add comment to capture "For button default use case, keep content as undefined and rely on Button label and icon props instead"

File: src/js/components/Menu/Menu.js [hunk 6]
```diff
@@ -210,9 +220,12 @@ const Menu = forwardRef((props, ref) => {
         active={activeItemIndex === controlButtonIndex}
         focusIndicator={false}
         hoverIndicator="background"
+        label={theme.button.default && !children ? menuLabel : undefined}
```
Comment (taysea) [inline src/js/components/Menu/Menu.js:223]: extract logic sure to separate condition --> and name single variable

Comment (taysea) [reply src/js/components/Menu/Menu.js:223]: to make more readable

File: src/js/components/Menu/Menu.js [hunk 7]
```diff
@@ -210,9 +220,12 @@ const Menu = forwardRef((props, ref) => {
...
+        icon={theme.button.default && !children ? menuIcon : undefined}
         plain={plain}
         onClick={onDropClose}
         onFocus={() => setActiveItemIndex(controlButtonIndex)}
+        reverse={theme.button.default && !children ? true : undefined}
```
Comment (taysea) [inline src/js/components/Menu/Menu.js:228]:
```suggestion
        reverse={theme.button.default && !children}
```
try this

Comment (taysea) [reply src/js/components/Menu/Menu.js:228]: Pull out these 3 variables and any other shared props and spread across controlMirror and DropButton

**Review by ShimiSun — APPROVED** [review]
Thanks for the clean-ups, much more readable. LGTM

**Review by halocline — APPROVED** [review]
Good to go.

File: src/js/components/Menu/Menu.js [hunk 8]
```diff
@@ -184,20 +184,40 @@ const Menu = forwardRef((props, ref) => {
...
-      pad={theme.button.default ? undefined : 'small'}
-      gap={label && icon !== false ? 'small' : undefined}
-    >
-      <Text size={size}>{label}</Text>
-      {icon !== false
-        ? (icon !== true && icon) || <MenuIcon color={iconColor} size={size} />
-        : null}
-    </Box>
-  );
+  const menuLabel = <Text size={size}>{label}</Text>;
```
Comment (ericsoderberghp) [inline src/js/components/Menu/Menu.js:187]: Can we be using `<Button size={size} label={label} />` instead of using `<Text />` for the label?

File: src/js/components/Menu/Menu.js [hunk 9]
```diff
@@ -184,20 +184,40 @@ const Menu = forwardRef((props, ref) => {
...
+  const menuIcon =
+    icon !== false
+      ? (icon !== true && icon) || <MenuIcon color={iconColor} size={size} />
+      : null;
+
+  const isDefaultButton = theme.button.default && !children;
```
Comment (ericsoderberghp) [inline src/js/components/Menu/Menu.js:193]: This is a minor comment...

Can we use `let buttonProps;` and then assign it in the same way we assign `content`? That would have a single source for the conditional logic.

File: src/js/components/Menu/Menu.js (hunk 8 above)
Comment (taysea) [reply src/js/components/Menu/Menu.js:187]: Good thought, let me have a look.

**Review by ericsoderberghp — APPROVED** [review]
